### ISSUE TITLE
[BUG] Forecasted values changing after reexecution. Reproducibility error

### ISSUE THREAD

**Issue by math-kao** [felipeangelimvieira/prophetverse#306]
## DESCRIPTION: 
Not possible to reproduce forecasted values. 
I tried to explicitly pass the PRNGKeys in the inference_engine and in the class Prophetverse model (Just for double check the issue, once the seed/prng key is by default jax.random.PRNGKey(0)

(hyphothesis) Error occurs when reusing the .fit() method, maybe the PRNGKey defined in the prophetverse class is not being able to reach the .fit() method.

Previous hyphothesis:
- Features values
- Specific effect that generates this behaviour

See the example below, where I used dataset loaded from the prophetverse lib and a simple model.
After restart the kernel in my local environment, the outputs (y_pred) value changed.

## CODE EXAMPLE :

```
from prophetverse.datasets._mmm.dataset1 import get_dataset

yy,Xx,lift_test,true_effect,model2= get_dataset()

model337 = Prophetverse(
    trend=PiecewiseLogisticTrend(
        changepoint_prior_scale=0.1,
        changepoint_interval=8,
        changepoint_range=-8,
    ),
    exogenous_effects=[
        (
            "seasonality",
            LinearFourierSeasonality(
                sp_list=["YE"],
                fourier_terms_list=[1],
                freq="Q",
                prior_scale=0.1,
                effect_mode="multiplicative",
            ),
            None,
        )
    ],
  inference_engine = MAPInferenceEngine(
            optimizer=LBFGSSolver(
                memory_size=1000,
                max_linesearch_steps=1000,
                learning_rate=0.001
    )))

model337.fit(y=yy,X=Xx)

fh = yy.index.get_level_values(-1).unique()
y_pred = model337.predict(X=Xx,fh=fh)
y_pred
```

**Comment by math-kao** [comment]
!! 
It also worked for me ! 

If is a hash seed problem, maybe codes using dicts/sets could be affecting the calculations/processes.

I find in the src/prophetverse/_model.py,
    if exogenous_effects is not None:
            for exog_effect_name, exog_effect in exogenous_effects.items():
...

I am not certain if this could be affecting the internal process of numpyro, but now being able to replicate results is way to good!
Thanks for the support

**Comment by felipeangelimvieira** [comment]
Re-opening so that we can properly track the source of the problem
